source/common/access_log/access_log_impl.cc

```diff
@@ -262,15 +262,17 @@ MetadataFilter::MetadataFilter(const envoy::config::accesslog::v3::MetadataFilte
     : default_match_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(filter_config, match_if_key_not_found, true)),
       filter_(filter_config.matcher().filter()) {
 
-  auto& matcher_config = filter_config.matcher();
+  if (filter_config.has_matcher()) {
```

Comment:
Should we add an explicit test for this case?

Reply:
Yes, definitely -- done